### PR TITLE
TEP-0114: Updating runs.md Documentation

### PR DESCRIPTION
**Pull request by vsinghai**
# Changes
Prior to this commit, the previous iteration
of the runs.md doc had some fields missing from
the API and the level of detail and clarification.

In this commit, we aim to clarify the API and update
it to make sure it is clear, upto date, and has useful
examples.

Fixes issue [#5158](https://github.com/tektoncd/pipeline/issues/5158)

/cc @jerop
/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/runs.md [hunk 1]
```diff
@@ -54,12 +54,16 @@ A `Run` definition supports the following fields:
 - Optional:
   - [`params`](#specifying-parameters) - Specifies the desired execution
     parameters for the custom task.
+  - [`status`](#) - Used for cancelling a `Run`.
```
Comment (lbernick) [inline docs/runs.md:57]: I think a few of these links are broken.

File: docs/runs.md [hunk 2]
```diff
@@ -54,12 +54,16 @@ A `Run` definition supports the following fields:
...
+  - [`statusMessage`](#) - Status message for cancellation.
+  - [`retries`](#) - Specifies propagating retries count to custom tasks.
```
Comment (lbernick) [inline docs/runs.md:59]:
```suggestion
  - [`retries`](#) - Specifies the number of times the `Run` should retry, if the custom task controller supports retries.
```

**Review by lbernick — COMMENTED** [review]
Thanks Varun! This is great to have, but I don't think we should close #5158 when this is merged (I think "fixes #" in your PR description will [result in that happening](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)).

I think there's a lot more documentation for runs that would be nice to have, such as realistic examples of ci/cd pipelines containing runs and links to useful existing custom tasks. @jerop maybe worth updating the issue with some ideas, and possibly @geriom you might have some thoughts here as well? Of course, out of scope for this PR.

**Comment by vsinghai** [comment]
Thanks for the feedback @lbernick! There is still currently a lot that I have left to do for this PR, including (but not limited to) having examples for the custom fields and providing more clarity to the API. The reason I created a PR was so that @jerop can just look over the incremental progress that I am making to make sure we are on the same path regarding how to fix #5158 . I apologize for the confusion.

**Comment by lbernick** [comment]
no problem, thanks for the explanation! I think creating a draft PR can be useful in situations like these- it lets reviewers know that it's still a WIP (hold does this to some extent but I wasn't sure why the PR was on hold)

**Pull request closed without merging**